packages/blockchain/src/index.ts

```diff
@@ -1564,4 +1564,18 @@ export default class Blockchain implements BlockchainInterface {
       return false
     }
   }
+
+  /**
+   * Helper to deterimine if a signer is in or out of turn.
+   * @param signer
+   * @param blockNumber
+   */
+  cliqueSignerInTurn(signer: Address, blockNumber: BN): boolean {
```

Comment:
Ah, I think it would make sense if we use the head of the chain as the block number here?

Reply:
(and actually for all other cases it generally makes no sense since `cliqueActiveSigners()` is only providing the active signer state of the latest block, just seeing on a second look)

Reply:
ah you are right.

I think I will make blockNumber optional and fall back default to head of the chain since the miner actually needs head+1, so it would be nice to pass in that exact number for my intended usage.

I will add a big warning to the param typedoc though that this is only based on the blockchain's current state of active signers. That's definitely important to understand when using the fn.

Reply:
Hmm, I have the impression that I can't yet follow parts of the argumentation.

So, the miner use case is the same as the "normal" tip of the chain use case (so: latest header + 1), or am I missing something?

So for this to work we could just completely drop the `blockNumber` input, do an async function and use the `getLatestHeader()` call as you suggested?

I am generally not a fan of doing this large warning, this is just not consistent. So I think we should rather do this limited version suggested above or rather go the whole round and read the active signers directly from `_cliqueLatestSignerStates` respectively iterate through that and match with the block number provided (would still be a fan of an additional option to leave `blockNumber` out). Then we should maybe also make `CLIQUE_SIGNER_HISTORY_BLOCK_LIMIT` public and refer to this constant in the code docs and point out that the function can only be called back by this block number range.

I guess this second version would be the most flexible implementation and rather covering the broadest range of eventual use cases - we should craft a bit properly here since we'll add this to the official library API.

Ok with both versions though, we can also start with the very simple one since the extended version can also be added later without being a breaking change.

Reply:
ok, thanks for explaining your viewpoint a bit more :) makes a lot of sense, I will just do the simpler version for now and once we have the need for looking back in history at different block numbers we can add that functionality. updated in https://github.com/ethereumjs/ethereumjs-monorepo/pull/1444/commits/dbc215e228466bdc8a6786695aaf0fed26e857cf

Reply:
Thanks for your flexibility here. 😛